### PR TITLE
Fix sanitycheck and other patches

### PR DESCRIPTION
**Pull request by finikorg**
@jhedberg
@dcpleung

### CODE REVIEW

**Comment by finikorg** [comment]
> posix, cpp, and newlib should not be ignored. Especially cpp and newlib.

Can you pass sanitycheck with these?

**Comment by dcpleung** [comment]
> > posix, cpp, and newlib should not be ignored. Especially cpp and newlib.
> 
> Can you pass sanitycheck with these?

Yes, 120 tests built without issue after fixing the `__packed` in bootloader.

**Comment by finikorg** [comment]
> > > posix, cpp, and newlib should not be ignored. Especially cpp and newlib.
> > 
> > 
> > Can you pass sanitycheck with these?
> 
> Yes, 120 tests built without issue after fixing the `__packed` in bootloader.

Just rerun those tests and they are failing. Can you try to run with

$ sanitycheck -p up_squared_adsp

**Comment by finikorg** [comment]
@dcpleung For example running newlib test case:

```
$ sanitycheck -p up_squared_adsp -t newlib
...
Building initial testcase list...
1 test configurations selected, 710 configurations discarded due to filters.
Adding tasks to the queue...

up_squared_adsp           tests/lib/mem_alloc/libraries.libc.newlib          FAILED: Build failure

	see: sanity-out/up_squared_adsp/tests/lib/mem_alloc/libraries.libc.newlib/build.log
total complete:    1/   1  100%  skipped:    0, failed:    1
0 of 1 tests passed (0.00%), 1 failed, 0 skipped with 0 warnings in 20.51 seconds
In total 722 test cases were executed on 1 out of total 214 platforms (0.47%)
...

```
etc


> > > > posix, cpp, and newlib should not be ignored. Especially cpp and newlib.
> > > 
> > > 
> > > Can you pass sanitycheck with these?
> > 
> > 
> > Yes, 120 tests built without issue after fixing the `__packed` in bootloader.
> 
> Just rerun those tests and they are failing. Can you try to run with
> 
> $ sanitycheck -p up_squared_adsp

Probably something is even more broken now, @jhedberg :
```
...
boards/xtensa/up_squared_adsp/bootloader/manifest.h:12:10: fatal error: toolchain.h: No such file or directory
   12 | #include <toolchain.h>
      |          ^~~~~~~~~~~~~
...
```

**Comment by finikorg** [comment]
For my tree here: https://github.com/thesofproject/zephyr/pull/35 I am able to run sanitycheck

**Comment by jhedberg** [comment]
@finikorg can you explain this toolchain.h issue to me? Doesn't the bootloader have the normal `include/` (in the Zephyr root) in its include path?

**Comment by finikorg** [comment]
> @finikorg can you explain this toolchain.h issue to me? Doesn't the bootloader have the normal `include/` (in the Zephyr root) in its include path?

Maybe you forgot to include include path to other components?
```
--- a/boards/xtensa/up_squared_adsp/bootloader/CMakeLists.txt
+++ b/boards/xtensa/up_squared_adsp/bootloader/CMakeLists.txt
@@ -4,11 +4,13 @@
 
 add_library(base_module base_module.c)
 target_include_directories(base_module PUBLIC
+  ${ZEPHYR_BASE}/include
   ${SOC_DIR}/${ARCH}/${SOC_PATH}/include
   )
 
 add_library(boot_module boot_module.c)
 target_include_directories(boot_module PUBLIC
+  ${ZEPHYR_BASE}/include
   ${SOC_DIR}/${ARCH}/${SOC_PATH}/include
   )
 
```

**Comment by jhedberg** [comment]
> Maybe you forgot to include include path to other components?

I forgot? 😄 Anyway, thanks for pointing that out. Does it build for you with that change?

**Comment by finikorg** [comment]
> > Maybe you forgot to include include path to other components?
> 
> I forgot? smile Anyway, thanks for pointing that out. Does it build for you with that change?

No, it does not build. Now it includes too much :)
```
include/toolchain/gcc.h:407:2: error: #error processor architecture not supported
  407 | #error processor architecture not supported
      |  ^~~~~
```

**Comment by finikorg** [comment]
> > > Maybe you forgot to include include path to other components?
> > 
> > 
> > I forgot? smile Anyway, thanks for pointing that out. Does it build for you with that change?
> 
> No, it does not build. Now it includes too much :)
> 
> ```
> include/toolchain/gcc.h:407:2: error: #error processor architecture not supported
>   407 | #error processor architecture not supported
>       |  ^~~~~
> ```

I think it is because we need to include configuration file which defines `CONFIG_XTENSA`

File: soc/xtensa/intel_apl_adsp/adsp.c [hunk 1]
```diff
@@ -98,8 +98,7 @@ static void send_fw_ready(void)
 	       &fw_ready_apl, sizeof(fw_ready_apl));
 
 	memcpy((void *)(MAILBOX_DSPBOX_BASE + sizeof(fw_ready_apl)),
-	       &sram_window,
-	       (sizeof(sram_window) + sram_window.ext_hdr.hdr.size));
+	       &sram_window, sizeof(sram_window));
```
Comment (finikorg) [inline soc/xtensa/intel_apl_adsp/adsp.c]: @dcpleung could you check this

**Comment by finikorg** [comment]
Closing outdated PR

**Pull request closed without merging**